pkg/apis/serving/fieldmask.go

```diff
@@ -183,6 +183,10 @@ func PodSpecMask(ctx context.Context, in *corev1.PodSpec) *corev1.PodSpec {
 	out.Volumes = in.Volumes
 	out.ImagePullSecrets = in.ImagePullSecrets
 	out.EnableServiceLinks = in.EnableServiceLinks
+	// Only allow setting AutomountServiceAccountToken to false
+	if !*in.AutomountServiceAccountToken {
```

Comment:
This'll panic because of a nil pointer deref, you'll want to do `if in.AutomountServiceAccountToken != nil && !*in.AutomountServiceAccountToken`